attestation/commandrun/commandrun.go

```diff
@@ -154,21 +225,46 @@ func (r *CommandRun) runCmd(ctx *attestation.AttestationContext) error {
 		enableTracing(c)
 	}
 
+	runtime.LockOSThread()
+
 	if err := c.Start(); err != nil {
 		return err
 	}
 
-	var err error
+	if r.tetragonAddress != "" {
+		syscall.Kill(c.Process.Pid, syscall.SIGSTOP)
```

Comment:
This seems racy -- the process may already well be off to the races by time it gets the SIGSTOP.  We likely want to do something similar to how we start the process with trace like [here](https://github.com/testifysec/go-witness/blob/06e9a338aa4a6f45de1c03c0b3adba0a1d271aad/attestation/commandrun/tracing_linux.go#L49) where the child process will call PTRACE_TRACEME before execve'ing to the new process.

We may not want to use TRACEME exactly since to get the process to go we'd have to attach and then detach a tracer to get the process to go after tetragon is up and running -- but we're going to need to modify all of this to make sure we're starting the trace after the new process is created but before execve is called.

Reply:
It looks like "TRACEME" is going to be what we want to use unless we want to rewrite the exec_linux package.  https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/syscall/exec_linux.go;drc=0a52d80666ddaa557cec17ad9166e2514b0bb6d4;l=62

Reply:
Correct, that's what we're setting for the ptrace tracing code I linked above.

To make that work we'd have to start the process with that attribute set, setup tetragon, and then attach and detach a ptracer to the process to get it to kick off.

The only other way around it would be to handle the forking and execve'ing ourselves.

Reply:
I don't want to write that fork/exec code for this feature -- seems like a can of worms.  I'll refactor your ptrace code to work with both tracers.

@mikhailswift  Do you think it would be useful to write a tracer interface?

Reply:
I can't say for sure but I have a hard time imagining an interface that would work well for both current tracing models we have since they differ so much.

At the moment just make the code that sets up the TRACEME syscall if either ptrace or tetragon tracing is setup and then brancing off to the correct path afterward is probably good enough.  For the ptrace code we will want to continue as normal and for the tetragon code we'll just want to detach after the first caught syscall.